test/src/device-state/current-state.spec.ts

```diff
@@ -505,25 +505,6 @@ describe('device-state/current-state', () => {
 			report.cancel();
 		});
 
-		it('does not report if current state has not changed', async () => {
-			// Use a temporary unhandledRejectionHandler to catch the promise
-			// rejection from the Bluebird.delay stub
-			process.on('unhandledRejection', unhandledRejectionHandler);
-			spy(_, 'size');
-
-			reportCurrentState();
-
-			// Wait 200ms for anonymous async IIFE inside reportCurrentState to finish executing
-			// TODO: is there a better way to test this? Possible race condition
-			await sleep(200);
-
-			expect(stateForReport).to.deep.equal({ local: {}, dependent: {} });
-			expect(_.size as SinonSpy).to.have.returned(0);
-
-			(_.size as SinonSpy).restore();
-			process.removeListener('unhandledRejection', unhandledRejectionHandler);
-		});
-
```

Comment:
I removed this case because it's incorrect for the reportCurrentState to decide if a state should be reported. That is implemented in the report function. Having both of these functions check if there is a change was redundant. This report function has the same test to make sure it's not reporting the same state https://github.com/balena-os/balena-supervisor/blob/a4be293ec48755f7c3c75a0db1db6bacb3f628d8/test/src/device-state/current-state.spec.ts#L407-L411

Reply:
I agree